drivers/vulkan/rendering_device_vulkan.cpp

```diff
@@ -8538,6 +8538,11 @@ void RenderingDeviceVulkan::_begin_frame() {
 			context->append_command_buffer(frames[frame].draw_command_buffer);
 			context->set_setup_buffer(frames[frame].setup_command_buffer); // Append now so it's added before everything else.
 		}
+
+		if (frames[frame].set_resources_initial_state) {
+			vkCmdResetQueryPool(frames[frame].setup_command_buffer, frames[frame].timestamp_pool, 0, max_timestamp_query_elements); //Reset all queries in a query pool before doing any operations with them.
```

Comment:
Couldn't we do this right after calling ``vkCreateQueryPool()``?

Reply:
We could do it with ```vkResetQueryPool()``` but that requires an extension to be enabled. We can't call```vkCmdResetQueryPool()``` at that point because the command buffer isn't in a recording state.

https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkResetQueryPoolEXT.html

The extension was promoted in Vulkan 1.2

Reply:
@clayjohn moved ```vkCmdResetQueryPool()``` after query pool creation.